velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -288,40 +300,54 @@ struct FromUnixtimeFunction {
     legacyFormatter_ = config.sparkLegacyDateFormatter();
     sessionTimeZone_ = getTimeZoneFromConfig(config);
     if (format != nullptr) {
-      setFormatter(*format);
+      if (!setFormatter(*format)) {
+        invalidFormat_ = true;
+      }
       isConstantTimeFormat_ = true;
     }
   }
 
-  FOLLY_ALWAYS_INLINE void call(
+  FOLLY_ALWAYS_INLINE bool call(
       out_type<Varchar>& result,
       const arg_type<int64_t>& second,
       const arg_type<Varchar>& format) {
+    if (invalidFormat_) {
+      return false;
+    }
     if (!isConstantTimeFormat_) {
-      setFormatter(format);
+      if (!setFormatter(format)) {
+        return false;
+      }
     }
     const Timestamp timestamp{second, 0};
     result.reserve(maxResultSize_);
     int32_t resultSize;
     resultSize = formatter_->format(
         timestamp, sessionTimeZone_, maxResultSize_, result.data(), true);
     result.resize(resultSize);
+    return true;
   }
 
  private:
-  FOLLY_ALWAYS_INLINE void setFormatter(const arg_type<Varchar>& format) {
-    formatter_ = detail::getDateTimeFormatter(
+  FOLLY_ALWAYS_INLINE bool setFormatter(const arg_type<Varchar>& format) {
+    auto formatter = detail::getDateTimeFormatter(
         std::string_view(format.data(), format.size()),
         legacyFormatter_ ? DateTimeFormatterType::STRICT_SIMPLE
                          : DateTimeFormatterType::JODA);
+    if (formatter.hasError()) {
+      return false;
+    }
```

Comment:
Exception from 'buildJodaDateTimeFormatter' is not handled, so it shall work here if 'buildJodaDateTimeFormatter' returns Expected.